pkg/utils/tls.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2023 Intel Corporation.  All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/viper"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// GetClientCredentials returns gRPC client credential based on user provided configuration.
+// if "--insecure=true" it will provide insecure.NewCredentials
+// if "--insecure=false" && "--mtls=true" it will provide mTLS credentials
+// otherwise only server side validation with CA added in client CA pool
+func GetClientCredentials() (credentials.TransportCredentials, error) {
+	if viper.GetBool("insecure") {
+		return insecure.NewCredentials(), nil
+	}
+
+	if viper.GetBool("mtls") {
+		return getClientMTLSCredentials()
```

Comment:
Instead of bool, if we have enums - 0 insecure 1 TLS 2 mTLS

Reply:
As discussed over call, this can be reviewed again as future improvement.